### PR TITLE
Room suggestions

### PR DESCRIPTION
**Pull request by mwang25277**
closes https://github.com/zednis/alexa-skill-meeting-scheduler/issues/38

### CODE REVIEW

File: api.js [hunk 1]
```diff
@@ -834,18 +873,69 @@ const createTimetable = function(obj) {
...
+                timeSlot.rooms.push(roomName);
+                for(var i = 0; i < meetingTimes.length - 1; i+=2) {
+
+                    var startTime = meetingTimes[i];
+                    var endTime = meetingTimes[i+1];
+
+                    if((((new Date(timeSlot.startDateTime)).getTime() <= (new Date(startTime)).getTime()) && 
+                    ((new Date(timeSlot.endDateTime)).getTime() > (new Date(startTime)).getTime())) ||
+                   (((new Date(timeSlot.startDateTime)).getTime() <= (new Date(endTime)).getTime()) && 
+                    ((new Date(timeSlot.endDateTime)).getTime() >= (new Date(endTime)).getTime()))) {
```
Comment (mwang25277) [inline api.js]: Not sure if these inequalities are correct

Comment (zednis) [reply api.js]: there is a lot of duplicate calls of Date().getTime() that we should be able to reduce.

IIRC the pseudocode should look like
```
if((meetingStart >= timeSlotStart && meetingStart <= timeSlotEnd) 
|| (meetingEnd > timeSlotStart && meetingEnd <= timeSlotEnd) 
|| (meetingStart <= timeSlotStart && meetingEnd >= timeSlotEnd)) {
// slot not available
} else {
  // slot available for this meeting
}
```

File: api.js [hunk 2]
```diff
@@ -802,19 +813,47 @@ const createTimetable = function(obj) {
...
-            if(((new Date(meetings[x].start_datetime)).getTime() <= (new Date(currDate.toISOString())).getTime()) && 
-               ((new Date(meetings[x].end_datetime)).getTime() > (new Date(currDate.toISOString())).getTime())) {
-              timetable[currDate.toISOString()] = 1;
+            //TODO: check inequality signs/logic
+            if((((new Date(meetings[x].start_datetime)).getTime() <= (new Date(currDate.toISOString())).getTime()) && 
+                ((new Date(meetings[x].end_datetime)).getTime() > (new Date(currDate.toISOString())).getTime())) ||
+               (((new Date(meetings[x].start_datetime)).getTime() <= (new Date(endDateTime.toISOString())).getTime()) && 
+                ((new Date(meetings[x].end_datetime)).getTime() >= (new Date(endDateTime.toISOString())).getTime()))) {
+              //timetable[timetable.length - 1].available = 1;
+                timeslot.available = 1;
```
Comment (zednis) [inline api.js]: As per the comment below, I think you should simplify this code by using variable assignment prior to the conditional.

```
const meetingStartTime = ((new Date(meetings[x].start_datetime)).getTime();
```

File: api.js (hunk 1 above)
Comment (zednis) [reply api.js]: another way to refactor would be to put this conditional into a function that returns a boolean.  then we can evaluate the conditional on its own with test data.  This is a really good way to unit test complex conditionals; separating the conditional from the logic that is utilizing it.

**Pull request merged**